dask/array/core.py

```diff
@@ -1181,19 +1181,33 @@ def to_dask_dataframe(self, columns=None):
         from ..dataframe import from_dask_array
         return from_dask_array(self, columns=columns)
 
-    def __int__(self):
-        return int(self.compute())
-
     def __bool__(self):
-        return bool(self.compute())
+        if self.size > 1:
+            raise ValueError("The truth value of a {0} is ambiguous. "
+                             "Use a.empty, a.bool(), a.item(), a.any() or a.all()."
+                             .format(self.__class__.__name__))
+        else:
+            return bool(self.compute())
```

Comment:
Any reason not to use `_scalarfunc` here?

Reply:
Only because I wanted a different exception and message for `__bool__` than was used in the others.

Reply:
Ok, makes sense. Thanks for the clarification.